### PR TITLE
MINOR: Correct logging and Javadoc in FetchSessionHandler

### PR DESCRIPTION
**Pull request by adamkotwasinski**
While debugging some client issues I found out that some Javadocs in FetchSessionHandler got misleading / outdated.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java [hunk 1]
```diff
@@ -454,7 +454,7 @@ String verifyFullFetchResponsePartitions(Set<TopicPartition> topicPartitions, Se
      * @param topicPartitions  The topicPartitions from the FetchResponse.
      * @param ids              The topic IDs from the FetchResponse.
      * @param version          The version of the FetchResponse.
-     * @return                 True if the incremental fetch response partitions are valid.
+     * @return                 null if the incremental fetch response partitions are valid; human-readable problem description otherwise.
```
Comment (showuon) [inline clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java:457]: Thanks for the nice catch! I'm wondering if the L467 and L469 are wrong. Currently, we did:
```java
       if (extra.isEmpty())
            bld.append("extraPartitions=(").append(Utils.join(extra, ", ")).append("), ");
        if (extraIds.isEmpty())
            bld.append("extraIds=(").append(Utils.join(extraIds, ", ")).append("), ");
```
And that will only append **empty** extras and extraIds, which is not right. Do you think it should be non-empty check?

File: clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java [hunk 2]
```diff
@@ -420,7 +420,7 @@ private String topicIdPartitionsToLogString(Collection<TopicIdPartition> partiti
      * @param topicPartitions  The topicPartitions from the FetchResponse.
      * @param ids              The topic IDs from the FetchResponse.
      * @param version          The version of the FetchResponse.
-     * @return                 True if the full fetch response partitions are valid.
+     * @return                 null if the fetch response partitions are valid; human-readable problem description otherwise.
```
Comment (showuon) [inline clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java]: I would keep the `full` word as before. Any reason why we should remove it?

**Review by showuon — COMMENTED** [review]
@adamkotwasinski , thanks for the PR. Left some comments. Thanks.

File: clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java (hunk 1 above)
Comment (jolshan) [reply clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java:457]: These should be non-empty checks. Thanks for catching.

File: clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java (hunk 2 above)
Comment (jolshan) [reply clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java]: +1 to this

Comment (adamkotwasinski) [reply clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java]: Ah! For the whole time I parsed it as `full (fetch response partitions)` and not `(full fetch) response partitions`, as the requests are named `Fetch` vs `IncrementalFetch`. Will add it back.

File: clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java (hunk 1 above)
Comment (adamkotwasinski) [reply clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java:457]: Yeah, that should be safe, the built string is anyways always returned if either of them is non-empty. Amended.

File: clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java (hunk 2 above)
Comment (jolshan) [reply clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java]: We can make this clearer if it helps. Maybe something like FullFetch? Not sure if there is a better option

Comment (adamkotwasinski) [reply clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java]: I think we are fine - the implication comes from method signature `verifyFullFetchResponsePartitions` and its Javadoc `Verify that a full fetch response...`

**Review by jolshan — APPROVED** [review]
lgtm, thanks for fixing!

**Review by dajac — APPROVED** [review]
LGTM

**Review by showuon — APPROVED** [review]
LGTM! Thanks for the fix.

**Comment by showuon** [comment]
Triggering jenkins build again to check the build results.
https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11843/3/

**Comment by dajac** [comment]
@showuon Can we merge this one?

**Comment by showuon** [comment]
Failed tests are unrelated.
```
    Build / ARM / kafka.network.SocketServerTest.testClientDisconnectionUpdatesRequestMetrics()
    Build / PowerPC / org.apache.kafka.streams.integration.NamedTopologyIntegrationTest.shouldRemoveAndReplaceTopologicallyIncompatibleNamedTopology
    Build / JDK 17 and Scala 2.13 / org.apache.kafka.connect.integration.BlockingConnectorTest.testBlockInSinkTaskStart
    Build / JDK 17 and Scala 2.13 / org.apache.kafka.connect.integration.ConnectWorkerIntegrationTest.testBrokerCoordinator
    Build / JDK 17 and Scala 2.13 / kafka.admin.DescribeConsumerGroupTest.testDescribeOffsetsOfExistingGroupWithNoMembers()
    Build / JDK 17 and Scala 2.13 / kafka.admin.LeaderElectionCommandTest.[1] Type=Raft, Name=testElectionResultOutput, Security=PLAINTEXT
    Build / JDK 17 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()
    Build / JDK 17 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()
    Build / JDK 11 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()
    Build / JDK 11 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()
```

**Pull request merged**